modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/CurrencyResourceImpl.java

```diff
@@ -141,7 +158,9 @@ public Currency postCurrency(Currency currency) throws Exception {
 
 		return _toCurrency(
 			_commerceCurrencyService.addCommerceCurrency(
-				currency.getExternalReferenceCode(), currency.getCode(),
+				GetterUtil.getString(
+					currency.getExternalReferenceCode(), currency.getCode()),
```

Comment:
replace with
`currency.getExternalReferenceCode()`
if externalReferenceCode is null it is generated